From ee1dfadec7be7c64a2bc2b34a1e4195b9048c46f Mon Sep 17 00:00:00 2001
From: Jan Janssen <medhefgo@web.de>
Date: Thu, 18 Aug 2022 13:41:49 +0200
Subject: [PATCH] boot: Add log_trace debugging helper

(cherry picked from commit 5966c54df4668abc17ae12c40fb0c30d31e80998)

Related: RHEL-16952
---
 src/boot/efi/log.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/boot/efi/log.h b/src/boot/efi/log.h
index f24034fd78..9bdcfad923 100644
--- a/src/boot/efi/log.h
+++ b/src/boot/efi/log.h
@@ -8,3 +8,4 @@ _gnu_printf_(2, 3) EFI_STATUS log_internal(EFI_STATUS status, const char *format
 #define log_error_status(status, ...) log_internal(status, __VA_ARGS__)
 #define log_error(...) log_internal(EFI_INVALID_PARAMETER, __VA_ARGS__)
 #define log_oom() log_internal(EFI_OUT_OF_RESOURCES, "Out of memory.")
+#define log_trace() log_internal(EFI_SUCCESS, "%s:%i@%s", __FILE__, __LINE__, __func__)
